From 29ef7608aeec673512168d716bedde8691a9baaa Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Mon, 11 May 2015 17:21:27 +0530
Subject: [PATCH 103/124] drivers:usb:fsl: Correct endpoint name checking

Endpoint name being statically defined cannot be NULL
hence checking name's first two character with "ep" for
endpoint validity

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Change-Id: I1fc6b71fce4807fdc18a96db0182c15dd9e3c649
Reviewed-on: http://git.am.freescale.net:8181/36196
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Reviewed-by: Honghua Yin <Hong-Hua.Yin@freescale.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index 5126e3a..7ca8219 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -1694,7 +1694,7 @@ static void dtd_complete_irq(struct fsl_udc *udc)
 		curr_ep = get_ep_by_pipe(udc, i);
 
 		/* If the ep is configured */
-		if (curr_ep->name == NULL) {
+		if (strncmp(curr_ep->name, "ep", 2)) {
 			WARNING("Invalid EP?");
 			continue;
 		}
-- 
2.1.0.27.g96db324

